spring-cloud-config-client/src/main/java/org/springframework/cloud/bootstrap/config/PropertySourceBootstrapConfiguration.java

```diff
@@ -143,12 +142,23 @@ public ConfigClientProperties configClientProperties() {
 		}
 
 		@Bean
+		@ConditionalOnMissingClass(name = {"org.springframework.cloud.config.server.EnvironmentRepository", "org.springframework.retry.RetryOperations"})
```

Comment:
I'm not sure if this is what we had in mind for disabling the PropertySourceLocator during server bootstrap phase.